### PR TITLE
@dynamicCompileEmit attribute

### PR DESCRIPTION
**Pull request by Hardcode84**
Will be useful with `bind`

### CODE REVIEW

**Comment by Hardcode84** [comment]
I can usually merge jit-related changes by myself (no one understands what's going on there anyway) but this is user-facing change so some review would be welcome (at least spelling).

Ping @JohanEngelen @kinke 

Thanks

**Comment by Hardcode84** [comment]
ping

File: src/ldc/attributes.d [hunk 1]
```diff
@@ -118,6 +118,28 @@ private struct _dynamicCompileConst
 {
 }
 
+/++
+ + When applied to a function, marks this function for dynamic compilation.
+ + Static version of function will be called via direct call but function
+ + body will still be available for jit facilities (e.g. jit bind).
```
Comment (kinke) [inline src/ldc/attributes.d]: I'd probably prefer `@dynamicCompilable`, I find it simpler to grasp ('can be compiled at runtime too').

**Comment by JohanEngelen** [comment]
I tried to improve the documentation text.

kinke wrote:
> I'd probably prefer @dynamicCompilable, I find it simpler to grasp ('can be compiled at runtime too').

I also think the "indirect" is confusing. But "dynamicCompilable" is very similar to "dynamicCompile", can we come up with a suffix after "dynamicCompile" ? (or will things become very verbose then...?)

**Comment by Hardcode84** [comment]
`@dynamicCompileEmit`?

**Comment by JohanEngelen** [comment]
"Emit" sounds ok.

**Comment by Hardcode84** [comment]
Updated, thanks.

**Comment by JohanEngelen** [comment]
Reminder: after merging, https://wiki.dlang.org/LDC-specific_language_changes#Attributes needs updating (also add LDC version that supports this attrib)

**Comment by Hardcode84** [comment]
I will merge this tomorrow if there are no objections.

**Pull request merged**